File: 2894. Divisible and Non-divisible Sums Difference/differenceOfSums.js
/**
 * @param {number} n
 * @param {number} m
 * @return {number}
 */
var differenceOfSums = function (n, m) {
    let num1 = 0;   //Sum of integers that not divisable by m
    let num2 = 0;   //Sum of integer that divisable by m

    for (let i = 1; i <= n; i++) {
        if (i % m == 0) {  //divisable
            num2 += i
        }
        else if (i % m !== 0) {   //Not divisable
            num1 += i
        }
    }

    // console.log("Num1 " , num1 , " Num2 " , num2 , " subrtact " , num1-num2)
    return num1 - num2;
};

File: 2894. Divisible and Non-divisible Sums Difference/differenceOfSums.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
    public:
        int differenceOfSums(int n, int m) {
            int num1 = 0;   //divisable by m
            int num2 = 0;   //not divisable by m
            for(int i=1 ; i<=n ;i++){
                if(i % m == 0){
                    num1+= i;
                }
                else{
                    num2+= i;
                }
            }
    
            return num2 - num1;
        }
    };